abuild: default_dbg: do not overwrite variables

tcely opened the pull request:
srcdir is very important for abuild operation.

Also, quoted various paths and contain directory changes to a subshell

ncopa commented on abuild.in:
```diff
@@ -1663,25 +1663,26 @@ doc() {
 # predefined splitfunc dbg
 default_dbg() {
-	local f
+	local _binfiles f _srcdir _srcfile _dstdir _dstfile
```
we cannot use identifiers prefixed with underscore in abuild.

ncopa reviewed (changes requested):
rename variable names to something without underscore prefix.

tcely commented:
@ncopa renamed variables as requested

tcely committed:
abuild: default_dbg: eliminate side effects

- do not overwrite variables

srcdir is very important for abuild operation

- quoted various paths
- use a sub-shell to contain directory changing

Resolves alpinelinux/abuild#58
